assets/src/components/propertiesPanel/blockWaiverBanner.tsx

```diff
@@ -22,77 +24,120 @@ const currentFuturePastClass = (blockWaiver: BlockWaiver): string => {
   }
 }
 
-const currentFuturePastTitle = (blockWaiver: BlockWaiver): string => {
+const Badge = ({
+  text,
+  icon,
+  modifier,
+}: {
+  text: string
+  icon?: ReactElement
+  modifier?: string
+}) => (
+  <div
+    className={`m-block-waiver-banner__badge ${
+      modifier ? "m-block-waiver-banner__badge--" + modifier : ""
+    }`}
+  >
+    {icon}
+    {text}
+  </div>
+)
+
+const BlockWaiverBadge = ({ blockWaiver }: { blockWaiver: BlockWaiver }) => {
+  const modifier = currentFuturePastClass(blockWaiver)
   switch (currentFuturePastType(blockWaiver)) {
-    case CurrentFuturePastType.Current:
-      return "Current"
-    case CurrentFuturePastType.Future:
-      return "Future"
     case CurrentFuturePastType.Past:
-      return "Past"
-  }
-}
+      return <Badge text={"Past"} modifier={modifier} />
 
-const alertIconStyle = (blockWaiver: BlockWaiver): AlertIconStyle => {
-  switch (currentFuturePastType(blockWaiver)) {
     case CurrentFuturePastType.Current:
-      return AlertIconStyle.Black
+      return (
+        <Badge
+          text={"Current"}
+          icon={
+            <span>
+              <IconAlertCircle style={AlertIconStyle.Black} />
+            </span>
+          }
+          modifier={modifier}
+        />
+      )
     case CurrentFuturePastType.Future:
-      return AlertIconStyle.GreyOnGrey
-    case CurrentFuturePastType.Past:
-      return AlertIconStyle.GreyOnGrey
+      return (
+        <Badge
+          text={"Upcoming"}
+          icon={
+            <NotificationBellIcon className="m-notification-bell-icon--read" />
+          }
+          modifier={modifier}
+        />
+      )
   }
 }
 
+const formattedReason = (blockWaiver: BlockWaiver) =>
+  [
+    blockWaiver.causeDescription,
+    (blockWaiver.remark || "")
+      .replace(blockWaiver.causeDescription, "")
+      .replace(/:$/, ""),
+  ]
+    .filter((s) => s !== null && s !== "")
+    .join(" ")
```

Comment:
**praise:** I appreciate that you also factored logic like this out while making your changes